src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -42,6 +42,7 @@ public ObjectResult(object value)
                 DeclaredType = DeclaredType,
                 ActionContext = context,
                 Object = Value,
+                StatusCode = StatusCode
```

Comment:
This change is moving the responsibility of setting the status code from the ObjectResult to the formatter in only one case. It doesn't seem right. Perhaps just a bool saying Status Code was set? (See the flow in Object result, it sets the status code first, then calls the formatter to act).

Btw when you look at the code here: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs#L55-L58

It's probaby a good idea to change to (to make it more readable and explicit)

```C#

if (StatusCode.HasValue)
{
    context.HttpContext.Response.StatusCode = StatusCode.Value;
}
```